src/02-common-principles.md

```diff
@@ -363,6 +363,102 @@ In particular, if a BIDS dataset contains a `derivatives/` sub-directory,
 the contents of that directory may be a heterogeneous mix of BIDS Derivatives
 datasets and non-compliant derivatives.
 
+## BIDS URI: Pointing to files within and outside of BIDS datasets
+
+Throughout the BIDS specification there a meta data fields that contain a
+pointer to a file.
+For example `IntendedFor`, `SpatialReference`, and `AssociatedEmptyRoom`.
+Prior to BIDS version 1.6.1, such pointers were required to be "relative paths".
+However the specification failed to consistently  describe relative to *what*
+these pointers should be specified, and the exact syntax to be used.
+Yet even with perfectly well and consistently defined "relative paths", this
+method of pointing to files becomes impractical when referring to files that
+are outside of a particular file tree (for example, on the Internet), or when
+the location of the data that is being pointed to changes.
+
+For these reasons, starting with BIDS version 1.6.1, using "relative paths" for
+all fields pointing to a separate file is [DEPRECATED][].
+
+Instead, the following [URI][uniform-resource-indicator] scheme SHOULD be
+used to point to files within and outside of BIDS datasets, as elaborated below.
+
+`bids:<dataset-name>:/absolute/path/within/dataset`
+
+Here, `bids:` is the URI scheme, and `<dataset-name>:/absolute/path/within/dataset`
+is the path to the file, consisting of the name of a dataset, a colon, and the absolute
+path within that dataset that MUST start with a forward slash `/`.
+
+The location of a dataset with the name `<dataset-name>` MUST be specified in the
+[`dataset_description.json` file](./03-modality-agnostic-files.md#dataset-description)
+under the `DatasetLinks` field, which is an [object][] of [strings][], as shown in
+the examples below.
+Note that the `<dataset-name>`: `"local"` is a reserved value that may only be
+used to refer to the current dataset.
+
+### Refer to a file within a dataset
+
+To link for example to a derivative file that is stored in a nested `derivatives/`
+directory within a dataset:
+`bids:deriv1:/sub-01/anat/sub-01_desc-preproc_T1w.nii.gz`
+
+```json
+{
+    "DatasetLinks": {
+        "deriv1": "file://derivatives/derivative1"
```

Comment:
How would I know that this is not a pointer to a `/derivatives/' directory in the root directory of my unix-like file system?

Reply:
let me try to answer myself. If the link is `file://` with two slashes, it is relative to the local dataset.

If the link is `file:///` with three slashes, it is relative to the root folder of the unix file system of the local computer. An (possibly easier to recognize) example would be `"deriv1": "file:///Users/roboos/megprojects/deriv1"`

I think the examples should be sorted from the easiest to recognize to more complex ones. An example that I would like to see is e.g. http://doi.org/10.18112/openneuro.ds003669.v1.0.0

Reply:
hmm, if it is relative to the local dataset, it should start with `local:` accordant to the description elsewhere, and not with `file:`. So I am still not sure.

Reply:
> let me try to answer myself. If the link is file:// with two slashes, it is relative to the local dataset.
If the link is file:/// with three slashes, it is relative to the root folder of the unix file system of the local computer. 

correct :+1: 

> hmm, if it is relative to the local dataset, it should start with local: accordant to the description elsewhere, and not with file:. So I am still not sure.

this is a difference between the BIDS URI (e.g., `bids:local:/some-dataset-root-relative-local-path`) and the location that is given as a mapping between a `<dataset-name>` and a location (e.g., `"deriv1": "http://doi.org/10.18112/openneuro.ds003669.v1.0.0"` for a remote location, or `"deriv2": "file:///home/stefanappelhoff/derivative2"` for a location on the host).

The reserved `local` keyword for `<dataset-name>` is a shorthand for specifying some other name (e.g., `derivX`) and then specifying that name within `dataset_description.DatasetLinks` as `"derivX": "file://...."`. That is convenient because if a dataset curator only ever uses `local` paths, then they don't have to define `DatasetLinks`.

Reply:
agreed. I think what confused me here is that for referring to a file within the present dataset there are two possibilities. You can use `local:` and give the file relative to the present dataset root, or you can use `file://` and give it the file relative to the present dataset root, (and third you could use `file:///` and give it relative to the filesystem root, but that is would not be portable).

in the 1st case the dataset name should not (or cannot?) be listed in `dataset_description.DatasetLinks`, whereas the 2nd case is specifically when it *is* listed there.

Reply:
yes!